### PR TITLE
Add RankGPT support inside RankLLM

### PR DESCRIPTION
**Pull request by xpbowler**
# Description

- Add RankGPT support inside RankLLM
- Add sliding window example to RankLLM
- Add parameter for choosing step_size
- Remove redundant example notebook in docs/docs/examples/node_postprocessor

### CODE REVIEW

File: docs/docs/examples/node_postprocessor/rankLLM.ipynb
Comment (logan-markewich) [inline docs/docs/examples/node_postprocessor/rankLLM.ipynb:1]: Any reason to remove (or move?) this notebook? (now it won't show up in the docs, maybe we should replace it with the one you added below?)

File: llama-index-integrations/postprocessor/llama-index-postprocessor-rankllm-rerank/llama_index/postprocessor/rankllm_rerank/base.py [hunk 1]
```diff
@@ -22,34 +24,43 @@ def __init__(
...
-        try:
-            model_enum = ModelType(model.lower())
-        except ValueError:
-            raise ValueError("Unsupported model type. Please use 'vicuna' or 'zephyr'.")
+        model_name = model.lower()
         from rank_llm.result import Result
 
         self._result = Result
 
-        if model_enum == ModelType.VICUNA:
```
Comment (logan-markewich) [inline llama-index-integrations/postprocessor/llama-index-postprocessor-rankllm-rerank/llama_index/postprocessor/rankllm_rerank/base.py:35]: Curious why we removed the enum? Seemed helpful for validation

**Review by logan-markewich — COMMENTED** [review]
A couple of nits, but looks good otherwise

File: llama-index-integrations/postprocessor/llama-index-postprocessor-rankllm-rerank/llama_index/postprocessor/rankllm_rerank/base.py (hunk 1 above)
Comment (xpbowler) [reply llama-index-integrations/postprocessor/llama-index-postprocessor-rankllm-rerank/llama_index/postprocessor/rankllm_rerank/base.py:35]: we'd need to add another parameter `gpt-type`, or continuously update the enums to match the available openai models. 

I wanted to keep the API clean. Tho thinking more, i think its ok adding another optional parameter, if that sounds good.

File: docs/docs/examples/node_postprocessor/rankLLM.ipynb
Comment (xpbowler) [reply docs/docs/examples/node_postprocessor/rankLLM.ipynb:1]: oh i see, i'll add it back

**Comment by xpbowler** [comment]
> A couple of nits, but looks good otherwise

@logan-markewich done!

**Review by nerdai — APPROVED** [review]
Can we also bump the version (patch) number in pyproject.toml?

**Pull request merged**